src/Marten/Events/Daemon/EventTypeFilter.cs

```diff
@@ -9,18 +12,19 @@ namespace Marten.Events.Daemon;
 /// <summary>
 ///     WHERE clause filter to limit event fetching to only the event types specified
 /// </summary>
-internal class EventTypeFilter: ISqlFragment
+internal sealed class EventTypeFilter: ISqlFragment
 {
     private readonly string[] _typeNames;
 
-    public EventTypeFilter(EventGraph graph, Type[] eventTypes)
+    public EventTypeFilter(EventGraph graph, IReadOnlyCollection<Type> eventTypes)
     {
         EventTypes = eventTypes;
-        _typeNames = eventTypes.Select(x => graph.EventMappingFor(x).Alias).ToArray();
+        // We need to load events that are mapped to this event type, not just the event itself.
+        var additionalAliases = graph.AliasesForEvents(eventTypes);
+        _typeNames = eventTypes.Select(x => graph.EventMappingFor(x).Alias).Union(additionalAliases).ToArray();
     }
 
-    public Type[] EventTypes { get; }
-
+    public IReadOnlyCollection<Type> EventTypes { get; }
```

Comment:
**QUESTION:** Could you expand on the change from Array to ReadonlyCollection?

Reply:
Just avoids the `ToArray` when initialized from the projection base.